torch/csrc/distributed/rpc/rpc_agent.h

```diff
@@ -130,7 +171,7 @@ class TORCH_API RpcAgent {
 
   // Stop accepting requests and shutdown the RPC framework as soon as possible
   // by terminating all RPC threads.
-  virtual void shutdown() = 0;
+  virtual void shutdown();
```

Comment:
From an API perspective for RpcAgent, it would be cleaner to introduce a protected pure virtual function called `shutdownImpl` that ProcessGroupAgent and ThriftAgent use to implement their shutdown mechanics. RpcAgent.shutdown() cleans up its state and then just calls `shutdownImpl`.